### ISSUE TITLE
Batch changes using `onlyFetchWorkspace` fail on dogfood

### ISSUE THREAD

**Issue by LawnGnome** [sourcegraph/sourcegraph#45915]
Haven't investigated this past seeing the failures yet, but it appears that workspaces with `onlyFetchWorkspace: true` always error with `Command failed with status -1` on k8s.sgdev.org at present.

Workspaces with `onlyFetchWorkspace: false` work as normal.

Example: https://k8s.sgdev.org/users/aharvey/batch-changes/is-this-thing-on/executions/QmF0Y2hTcGVjOiI3YjI5YjZlOC02ZDljLTRmNjYtYWJjZi1jNGY3YTY0MDMwMGMi/execution/workspaces/QmF0Y2hTcGVjV29ya3NwYWNlOjE5MzU3OTQ=

**Comment by eseliger** [comment]
```
getting diff produced by step: git add failed: 'git add --all' failed: fatal: 'github.com/sourcegraph/src-cli' does not appear to be a git repository
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
fatal: could not fetch de75d79e40449da3d266fa6aa1311b36fcd4589c from promisor remote: exit status 128 (exit code: 1)
```

I think that could be fixed by running git add --all in the subdirectory instead of the repo root - this is caused by the sparse checkouts we do.

**Comment by courier-new** [comment]
Thanks for the investigation. Assuming that's the fix, maybe this is 0.5d-1d? Again not really sure how we're going to prioritize non-strat-ready issues/bugs right now.

**Comment by eseliger** [comment]
I feel like bugs are always a similar priority as very-high if they are not too edge-casey. At least that's my personal take on that. 
Yeah probably more a 1d, since I'm not sure if that actually does anything good and needs some more investigation 😆

**Comment by courier-new** [comment]
Agreed, but it's also a lot easier to justify that when there's 5 engineers and the estimate for all bugs is <10% of their total time. The bugs start adding up pretty quickly if it's just me and Bolaji... 😅
